src/Common/src/TypeSystem/Common/CastingHelper.cs

```diff
@@ -317,6 +317,11 @@ private static bool CanCastToClass(this TypeDesc thisType, TypeDesc otherType, S
         {
             TypeDesc curType = thisType;
 
+            if (curType.IsInterface && otherType.IsObject)
```

Comment:
Could you please add a case for this to the casting unit tests as well? (src\ILCompiler.TypeSystem\tests\CastingTests.cs)